<a href="https://colab.research.google.com/github/Jubin5/Machine-Learning/blob/main/ML_Models(Breast_Cancer_dataset)_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
import pandas as pd
import numpy as np

# Load the breast cancer dataset
data = load_breast_cancer()
X = pd.DataFrame(data.data,columns=data.feature_names)
y = data.target

In [ ]:
X.isnull().sum() #Here in this dataset there is no missing values

mean radius                0
mean texture               0
mean perimeter             0
mean area                  0
mean smoothness            0
mean compactness           0
mean concavity             0
mean concave points        0
mean symmetry              0
mean fractal dimension     0
radius error               0
texture error              0
perimeter error            0
area error                 0
smoothness error           0
compactness error          0
concavity error            0
concave points error       0
symmetry error             0
fractal dimension error    0
worst radius               0
worst texture              0
worst perimeter            0
worst area                 0
worst smoothness           0
worst compactness          0
worst concavity            0
worst concave points       0
worst symmetry             0
worst fractal dimension    0
dtype: int64

In [ ]:
# Perform feature scaling
scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.2, random_state=42)


In [ ]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score
LR=LogisticRegression()
LR.fit(X_train,y_train)

LogisticRegression()

In [ ]:
from sklearn.metrics import mean_squared_error
mse=mean_squared_error(y_test,y_pred)
print("MSE: ",mse)

MSE:  0.02631578947368421


In [ ]:
from sklearn.tree import DecisionTreeClassifier
dt=DecisionTreeClassifier()
dt.fit(X_train,y_train)

DecisionTreeClassifier()

In [ ]:
y_pred=dt.predict(X_test)

In [ ]:
from sklearn.metrics import mean_squared_error
mse=mean_squared_error(y_test,y_pred)
print(f"MSE: {mse}")

MSE: 0.07017543859649122


In [ ]:
from sklearn.ensemble import RandomForestClassifier
rf=RandomForestClassifier()
rf.fit(X_train,y_train)


RandomForestClassifier()

In [ ]:
from sklearn.metrics import mean_squared_error
y_pred=rf.predict(X_test)
mse=mean_squared_error(y_test,y_pred)
print(f"MSE: {mse}")


MSE: 0.03508771929824561


In [ ]:
from sklearn.svm import SVC
svm=SVC()
svm.fit(X_train,y_train)

SVC()

In [ ]:
y_pred=svm.predict(X_test)
mse=mean_squared_error(y_test,y_pred)
print(f"MSE: {mse}")

MSE: 0.02631578947368421


In [ ]:
from sklearn.neighbors import KNeighborsClassifier
knn=KNeighborsClassifier()
knn.fit(X_train,y_train)

KNeighborsClassifier()

In [ ]:
y_pred=knn.predict(X_test)
mse=mean_squared_error(y_test,y_pred)
print(f"MSE: {mse}")

MSE: 0.05263157894736842


Out of the five  models we developed the best model is Logistic Regression and Support Vector Machine,because they both have the minimum mean squared error of 2.63%,And out of the five based on the error the worst one is decision tree which has 7% error.